include/klee/Expr.h

```diff
@@ -176,7 +176,26 @@ class Expr {
 
 protected:  
   unsigned hashValue;
-  
+
+  /// Compares `b` to `this` Expr for structural equivalence ignoring
+  /// kid expressions (i.e. those returned by `getKid()`).
+  /// Typically this means defining a total order over the internal attributes
+  /// of the Expr.
+  ///
+  /// Implementations can assume that `b` and `this` are of the same kind.
+  ///
+  /// This can be viewed as defining a partial order over Expr of the same
+  /// kind (partial because kid Expr are not compared).
+  ///
+  /// \param [in] b Expr to compare `this` to.
+  ///
+  /// \return One of the following values:
+  ///
+  /// * -1 if `this` is < `b` ignoring kid expressions.
+  /// * 0 if `this` is structurally equivalent to `b` ignoring kid expressions.
+  /// * 1 if `this` is > `b` ignoring kid expressions.
+  virtual int compareContents(const Expr &b) const = 0;
```

Comment:
The total order on the internal attributes is just a hint at how to implement the method. The fact this method in effect defines a partial order over Expr of the same kind is the more important point.

If it annoys you can I remove the mentioning total order.